### PR TITLE
Feature: Add middle mouse panning

### PR DESCRIPTION
**Pull request by milestone-dev**

### CODE REVIEW

**Review by TheNitesWhoSay — COMMENTED** [review]
Looks good to me feature-wise (I personally prefer the continuous scroll over the anchored position), would prefer not adding redundant public accessors/mutators for panning info only used in GuiMap/just use the values directly (unless you have some plan or reasoning I'm not seeing). Also KillTimer needs to occur on destruction of GuiMap as well as on MButtonUp if a timer is still live.

File: ChkdraftLib/Windows/MainWindows/GuiMap.h [hunk 1]
```diff
@@ -32,6 +32,13 @@ class GuiMap : public MapFile, public WinLib::ClassWindow, public IObserveUndos
                     bool setCurrPlayer(u8 newPlayer);
                     bool isDragging();
                     void setDragging(bool dragging);
+                    int getPanStartX();
```
[the remark this answers is not in the document]
Comment (milestone-dev) [reply ChkdraftLib/Windows/MainWindows/GuiMap.h]: Yeah, this is just me not knowing c++ and WinAPI well enough and just copying how other parts of this file was set up :)

File: ChkdraftLib/Windows/MainWindows/GuiMap.cpp [hunk 2]
```diff
@@ -17,7 +17,7 @@ GuiMap::GuiMap(Clipboard & clipboard, const std::string & filePath)
     bitmapHeight(0), bitmapWidth(0), currLayer(Layer::Terrain), currPlayer(0), zoom(1), RedrawMiniMap(true), RedrawMap(true),
     dragging(false), snapLocations(true), locSnapTileOverGrid(true), lockAnywhere(true),
     snapUnits(true), stackUnits(false), mapId(0), unsavedChanges(false), changeLock(false), undos(*this),
-    minSecondsBetweenBackups(1800), lastBackupTime(-1)
+    minSecondsBetweenBackups(1800), lastBackupTime(-1), panStartX(-1), panStartY(-1), panCurrentX(0), panCurrentY(0), panTimerID(0)
```
Comment (heinermann) [inline ChkdraftLib/Windows/MainWindows/GuiMap.cpp:20]: Default values can be set in the header file, would make this file more readable and reduce duplication. Probably something for a refactor.

Comment (TheNitesWhoSay) [reply ChkdraftLib/Windows/MainWindows/GuiMap.cpp:20]: indeed, default member initializers are generally better & I favor them in new code

**Review by TheNitesWhoSay — APPROVED** [review]

**Pull request merged**